### ISSUE TITLE
Single expression inside parens results in a block

### ISSUE THREAD

**Issue by mhanberg** [elixir-lang/elixir#13324]
### Elixir and Erlang/OTP versions

Erlang/OTP 26 [erts-14.2.1] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [jit:ns]

Elixir 1.17.0-dev (compiled with Erlang/OTP 26)

Main, as of a few days ago

### Operating system

Ubuntu mantic 23.10 x86_64

### Current behavior

With `Code.string_to_quoted`, the following code results in a `:__block__` instead of being a lone expression

```elixir
(!false)
```

becomes

```elixir
{:__block__, [], [{:!, [line: 1, column: 2], [false]}]}
```

I would have expected this to be

```elixir
{:!, [{:line, 1}, {:column, 2}], [false]}
```

If the current behavior is correct, what am I missing that makes it into a block?

### Expected behavior

I would have expected this to be

```elixir
{:!, [{:line, 1}, {:column, 2}], [false]}
```

If the current behavior is correct, what am I missing that makes it into a block?

**Comment by ericmj** [comment]
This is expected. Parentheses are used to support multiple expressions in blocks.

The optional syntax section in the docs explain it a bit further: https://hexdocs.pm/elixir/syntax-reference.html#optional-syntax

**Comment by mhanberg** [comment]
This is a single expression though.

The following does not produce block

```elixir
iex(9)> Code.string_to_quoted("(foo())")
{:ok, {:foo, [line: 1], []}}
```

it seems like the `not` and `!` operators cause the resulting AST to be wrapped in a `:__block__` with a single expression.

Is that expected? My previous understanding was that it would only be wrapped in a block if there was an expression list with more than one expression in it, meaning

```elixir
(
  foo = 1
  foo * 2
)

# or

(foo = 1; foo *2)
```

**Comment by sabiwara** [comment]
This seems to be done on purpose [here](https://github.com/elixir-lang/elixir/blob/7b73407/lib/elixir/src/elixir_parser.yrl#L1066-L1067).
Removing the clause "fixes" it but would remove the parentheses in this [formatter test case](https://github.com/elixir-lang/elixir/blob/7b73407/lib/elixir/test/elixir/code_formatter/operators_test.exs#L159). But maybe it should be addressed in the formatter instead?

**Comment by mhanberg** [comment]
Thanks for finding the spot in the parser!

I don't have strong feelings on whether it should be changed at all, but wanted to be clear on what the logic was that produced it.

So it seems like specifically if you have a set of parens with a single `!` or `not` expression, then it get wrapped in a block, correct?

**Comment by josevalim** [comment]
Yup, it is done for backwards compatibility, more info also here: https://github.com/elixir-lang/elixir/blob/7b73407/lib/elixir/src/elixir_parser.yrl#L741

It will be fixed on v2.0.

**Comment by mhanberg** [comment]
Thanks!
